### PR TITLE
Enable Readium to open File objects directly

### PR DESCRIPTION
**Pull request by johanpoirier**
Readium can now open a [File](https://developer.mozilla.org/en-US/docs/Web/API/File) object ([Blob](https://developer.mozilla.org/fr/docs/Web/API/Blob)) directly.

For example, if you're using the DeviceStorage API, you can read an epub file from the sdcard of a device :

    var sdcard = navigator.getDeviceStorage('sdcard');
    var request = sdcard.get("9782019111281.epub");
    request.onsuccess = function () {
        var file = this.result;
        var readium = new Readium({ jsLibRoot: './libs/' }, { el: '#reader-frame' });
            readium.openPackageDocument(file, function (packageDocument, options) {
                // do whatever you want with your packageDocument...
            });
    };

### CODE REVIEW

**Comment by danielweck** [comment]
Follow-up on this: as much as I love the idea of being able to open (zipped) EPUB from any data source that can ultimately provide a binary Blob (e.g. drag-n-drop, LocalStorage, SDcard, etc.), there are a number of serious caveats when using non-exploded/expanded EPUBs (regardless of the actual data source).

Documentation:
https://docs.google.com/document/d/1bwv89vSmLbUs4tHOwiqyeV546k5CG3RCJ1S6_-4wvvc/

In a nutshell, list of broken features:
- EPUB3 Media Overlays
- Any large video/audio HTML5 media
- Some SVG
- Any dynamically-injected resource (via scripting, i.e. not via static markup declarations)
- Dynamic hyperlinks
...basically, anything that can not easily be replaced with BlobURIs at parsing time (statically).

Thus the "low" priority on this issue / PR.

**Comment by danielweck** [comment]
@johanpoirier thank you very much for your PR! I adapted it and merged it to develop:
https://github.com/readium/readium-js/pull/115

**Comment by johanpoirier** [comment]
:+1:
